yt_dlp/extractor/vk.py

```diff
@@ -701,3 +704,142 @@ def _real_extract(self, url):
         return self.playlist_result(
             entries, post_id, join_nonempty(uploader, f'Wall post {post_id}', delim=' - '),
             clean_html(get_element_by_class('wall_post_text', webpage)))
+
+
+class VKPlayBaseIE(InfoExtractor):
+    _RESOLUTIONS = {
+        'tiny': '256x144',
+        'lowest': '426x240',
+        'low': '640x360',
+        'medium': '852x480',
+        'high': '1280x720',
+        'full_hd': '1920x1080',
+        'quad_hd': '2560x1440',
+    }
+
+    def _extract_initial_state(self, url, video_id):
+        webpage = self._download_webpage(url, video_id)
+        return self._search_json(
+            r'<script[^>]+\bid="initial-state"[^>]*>', webpage, 'initial state', video_id, fatal=False)
```

Comment:
Since this is fatal now, we can reduce some more code duplication
```suggestion
    def _extract_initial_state(self, url, video_id, path):
        webpage = self._download_webpage(url, video_id)
        initial_state = traverse_obj(self._search_json(
            r'<script[^>]+\bid="initial-state"[^>]*>', webpage, 'initial state', video_id),
            path, expected_type=dict)
        if not initial_state:
            raise ExtractorError('Unable to extract initial state')
        return initial_state
```